File: app/src/main/java/hexlet/code/games/CalcGame.java
package hexlet.code.games;

import java.util.Random;

public class CalcGame extends Game {

    private static final int MAX_NUMBER = 49;

    @Override
    public String getRule() {
        return "What is the result of the expression?";
    }

    @Override
    public GameRound getNewRound() {
        var random = new Random();

        var op = OPS.getRandom();
        var num1 = random.nextInt(1, MAX_NUMBER + 1);
        var num2 = random.nextInt(1, MAX_NUMBER + 1);

        var task = "%d %s %d".formatted(num1, op, num2);
        var calcResult = op.calc(num1, num2);
        var rightAnswer = Integer.toString(calcResult);

        return new GameRound(task, rightAnswer);
    }
}

File: app/src/main/kotlin/hexlet/code/CalcGame.kt
package hexlet.code

class CalcGame : Game() {
    override fun rule(): String {
        return "What is the result of the expression?"
    }

    override fun newRound(): GameRound {
        val op = operations.random()
        val n1 = (1..numberLimit).random()
        val n2 = (1..numberLimit).random()

        val question = question(op, n1, n2)
        val rightAnswer = rightAnswer(op, n1, n2)

        return GameRound(question, rightAnswer)
    }

    private fun question(op: String, n1: Int, n2: Int): String {
        return "Question: $n1 $op $n2"
    }

    private fun rightAnswer(op: String, n1: Int, n2: Int): String {
        val result = calculate(op, n1, n2)
        return result.toString()
    }

    private fun calculate(op: String, n1: Int, n2: Int): Int? {
        return when (op) {
            "+" -> n1 + n2
            "-" -> n1 - n2
            "*" -> n1 * n2
            "/" -> n1 / n2
            else -> null
        }
    }

    companion object {
        private final val operations = arrayOf("+", "-", "*")

        private const val numberLimit = 49
    }
}
